src/Std/Sat/AIG/Basic.lean

```diff
@@ -0,0 +1,510 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Henrik Böving
+-/
+import Std.Data.HashMap
+import Std.Data.HashSet
+
+namespace Std
+namespace Sat
+
+/-!
+This module contains the basic definitions for an AIG (And Inverter Graph) in the style of AIGNET,
+as described in https://arxiv.org/pdf/1304.7861.pdf section 3. It consists of an AIG definition,
+a description of its semantics and basic operations to construct nodes in the AIG.
+-/
+
+variable {α : Type} [Hashable α] [DecidableEq α]
+
+namespace AIG
+
+/--
+A circuit node. These are not recursive but instead contain indices into an `AIG`, with inputs indexed by `α`.
+-/
+inductive Decl (α : Type) where
```

Comment:
I don't think this needs changing, but is there a reason why you are not universe-polymorphic here?

Reply:
No particular reason.